### PR TITLE
tvision: new recipe.

### PR DESCRIPTION
**Pull request by OscarL**
TVision apps not looking 100% right, same issue as in `mc`: Terminal drawing leaving "vertical gaps" between "line/box drawing" chars.

Other than that, seems pretty functional.

Patched it to build/use shared libs.

The `turbo` editor (scintilla + tvision based), will follow after this one.

---

tested on beta4, both 32, and 64 bits.

### CODE REVIEW

File: dev-libs/tvision/tvision-2024.02.04~git.recipe [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	cmake --build ./build $jobArgs
+}
+
+INSTALL()
+{
+	cmake --install ./build
+
+	prepareInstalledDevelLib libtvision
+
+	sed -e 's|_IMPORT_PREFIX}/include|_IMPORT_PREFIX}/develop/headers|' -i \
```
[the remark this answers is not in the document]
Comment (augiedoggie) [reply dev-libs/tvision/tvision-2024.02.04~git.recipe:86]: There is a command from the simgear recipe that shows one way to do it.
```
sed -i \
	-e "s|\(\${_IMPORT_PREFIX}/\)include;|\1$relativeIncludeDir;|" \
	"$libDir"/cmake/SimGear/SimGearTargets.cmake
```

File: dev-libs/tvision/tvision-2024.02.04~git.recipe [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	cmd:tvhc$commandSuffix = $portVersion
+	"
+REQUIRES_tools="
+	$REQUIRES
+	tvision$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libncursesw$secondaryArchSuffix
```
Comment (Begasus) [inline dev-libs/tvision/tvision-2024.02.04~git.recipe:56]: As the test are run during BUILD it needs gtest also (both arch's), you could add it as disabled and mention that in the BUILD section also.

File: dev-libs/tvision/tvision-2024.02.04~git.recipe [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	cmd:cmake
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	# This is needed only when building the examples, as to binaries there need to run from the same
+	# "build" dir where the libtvsion.so and bins are placed.
+	export LIBRARY_PATH="%A:$LIBRARY_PATH"
```
Comment (Begasus) [inline dev-libs/tvision/tvision-2024.02.04~git.recipe:68]: This doesn't seem to do the trick here? The next one does though.
```
	export LIBRARY_PATH="$sourceDir/build${LIBRARY_PATH:+:$LIBRARY_PATH}"
```

Comment (Begasus) [reply dev-libs/tvision/tvision-2024.02.04~git.recipe:68]: Checked both arch's, test all good now.

File: dev-libs/tvision/tvision-2024.02.04~git.recipe [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+	cmake . -B ./build \
+		$cmakeDirArgs \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_SKIP_RPATH=YES \
+		-DTV_BUILD_EXAMPLES=YES \
+		-DTV_BUILD_TESTS=NO \
```
Comment (Begasus) [inline dev-libs/tvision/tvision-2024.02.04~git.recipe:75]: Loose trailing backslash:
```diff
-		-DTV_BUILD_TESTS=NO \
+		-DTV_BUILD_TESTS=NO
```

**Pull request merged**